src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp

```diff
@@ -4942,6 +4969,23 @@ void MusicXMLParserPass2::deleteHandledSpanner(SLine* const& spanner)
     delete spanner;
 }
 
+InferredPercInstr MusicXMLParserPass2::inferredPercInstr(const Fraction& tick, const staff_idx_t staffIdx)
+{
+    InferredPercInstr instr = InferredPercInstr();
+
+    for (InferredPercMap::iterator iter = m_inferredPerc.begin(); iter != m_inferredPerc.end();) {
```

Comment:
An iterative search like this defies the purpose of using a map. If you're scanning it, then you should store this as a simple vector (see my proposed solution b below). The power of using the map is that you can query it directly from the key (solution a). If you're expecting a large number of these, querying a map will be much faster. Otherwise it's whatever you prefer :)